Removed  legacy notification to connect with WooCommerce

octaedro opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR removes a legacy JITM for connecting with WooCommerce.com. This message is now a part of `WC Admin` bar via `Inbox`.

Closes #25889 .

### How to test the changes in this Pull Request:

1. Go to a WooCommerce wp-admin dashboard for a store that's not yet connected to WooCommerce.com
2. The JITM `Connect your store to WooCommerce.com to reveice extensions updates and support.` message shouldn't be visible.

### Other information:

* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

octaedro committed:
Removed  legacy notification to connect with WooCommerce

Removed legacy JITM for connecting with WooCommerce.com

claudiosanches commented on includes/admin/helper/class-wc-helper.php:
```diff
@@ -1527,12 +1527,6 @@ private static function _prompt_helper_connect( $screen_id ) {
 
 		$active_woo_plugins = array_intersect_key( $woo_plugins, array_flip( $active_plugins ) );
 
-		if ( count( $active_woo_plugins ) > 0 ) {
```
Now this function there's no use, but will keep running in `WC_Helper::admin_notices`.
Also this a private method, so we can just remove it.

octaedro replied:
Nice catch @claudiosanches! It's a good idea!
I removed the function in the commit  [134c0ea](https://github.com/woocommerce/woocommerce/pull/26167/commits/134c0ea0db8fd934053906f7c68beedf5af2ea30)

octaedro committed:
The function '_prompt_helper_connect' was deleted

After removing the legacy code the method '_prompt_helper_connect' wasn't useful anymore so it was removed.

claudiosanches reviewed (approved):
LGTM!
